tests/bench/harness.py

```
@@ -0,0 +1,683 @@
+            timeout: Maximum time to wait in seconds
+
+        Returns:
+            True if database is ready, False otherwise
+        """
+        print("Waiting for database to be ready...")
+        start_time = time.time()
+
+        while time.time() - start_time < timeout:
+            try:
+                # Test database connection using pg_isready
+                result = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)